src/common/procedure/src/local.rs

```diff
@@ -210,19 +209,28 @@ impl ManagerContext {
         procedures.contains_key(&procedure_id)
     }
 
+    /// Returns the number of running procedures.
+    fn num_running_procedures(&self) -> usize {
+        self.running_procedures.lock().unwrap().len()
+    }
+
     /// Try to insert the `procedure` to the context if there is no procedure
     /// with same [ProcedureId].
     ///
     /// Returns `false` if there is already a procedure using the same [ProcedureId].
     fn try_insert_procedure(&self, meta: ProcedureMetaRef) -> bool {
+        let procedure_id = meta.id;
         let mut procedures = self.procedures.write().unwrap();
-        if procedures.contains_key(&meta.id) {
+        if procedures.contains_key(&procedure_id) {
```

Comment:
The `insert` method returns None to indicate that the key is not present, so we don't need to call `contains_key`.